etna/records/views/records.py

```diff
@@ -87,11 +89,38 @@ def record_detail_view(request, id):
 
     page_title = f"Catalogue ID: {record.iaid}"
     image = None
+    iiif_manifest_url = None
 
     try:
-        iiif_manifest_url = iiif.manifest_url_for_record(record)
-    except iiif.RecordHasNoManifest:
-        iiif_manifest_url = None
+        # TODO: Use the contents from the request below to fetch the IIIF manifest
+        #       in order to build the HTML-only view (progressive enhancement).
+        #
+        #       Right now this is only used to establish if the record has
+        #       a IIIF manifest which could use a HEAD request instead.
+        #
+        #       We need to know if a record has IIIF manifest in order to establish
+        #       if we should load the IIIF viewer to the user. Not all records have
+        #       one.
+        #
+        #       We could make a HEAD request instead to check that, but it feels
+        #       short-sighted given that the progressively enhanced view will
+        #       require all this content to be fetched anyway.
+        #
+        #       This information could also be returned in the fetch endpoint
+        #       so we know in advance if we need to call the IIIF manifest
+        #       endpoint at all. The raised ticket:
+        #       https://national-archives.atlassian.net/browse/DOR-53
+        records_client.fetch_iiif_manifest(id=record.iaid)
```

Comment:
**Question:** Should this only run for certain record types? E.g. we might not want to make the request for `ARCHON` and `CREATEORS` record types. They also have different template which means that the viewer is not implemented there.

Reply:
Yes, I think we only want this for `digitised` records which cover all digitised records at TNA. I don't think ARCHON or CREATORS will have any images/digitised stuff. Not sure if it'd be easy to prevent it running on ARCHON etc?

Reply:
Just added an if-statement, same as for the template selection. An observation: It feels like the view as a function-based view looks quite messy given it does so much right now. There's a lot of noise all in one function. It may be beneficial to implement it as a class-based view or break down the function at some point.